lib/spack/spack/util/path.py

```diff
@@ -157,20 +160,23 @@ def substitute_config_variables(path):
 
     Spack allows paths in configs to have some placeholders, as follows:
 
-    - $env                 The active Spack environment.
-    - $spack               The Spack instance's prefix
-    - $tempdir             Default temporary directory returned by tempfile.gettempdir()
-    - $user                The current user's username
-    - $user_cache_path     The user cache directory (~/.spack, unless overridden)
-    - $architecture        The spack architecture triple for the current system
-    - $arch                The spack architecture triple for the current system
-    - $platform            The spack platform for the current system
-    - $os                  The OS of the current system
-    - $operating_system    The OS of the current system
-    - $target              The ISA target detected for the system
-    - $target_family       The family of the target detected for the system
-    - $date                The current date (YYYY-MM-DD)
-    - $spack_short_version The spack short version
+    - $env                  The active Spack environment.
+    - $spack                The Spack instance's prefix
+    - $tempdir              Default temporary directory returned by tempfile.gettempdir()
+    - $user                 The current user's username
+    - $user_cache_path      The user cache directory ($XDG_DATA_HOME, unless overridden)
+    - $spack_xdg_cache_home Backup location for temporary data
+    - $spack_xdg_state_home Long-lived but not-essential cache
```

Comment:
I find this name redirection confusing as a developer and a user. Can we map these names closer to what they are?

Like `misc_cache = $spack_xdg_state_home/$spack_instance/cache` could we call it `$misc_cache_base` instead? or just `state_home` if we want to keep the implied xdg mapping? 

If we want to explicitly tie xdg mapping to spack internal variables then I would propose doing it uniformly, but I think that will be a more disruptive change.

Reply:
I think the PR has moved a bit since this comment: the default settings have been restored, but with special default config values that respect XDG_/SPACK_ data home, also there is now just `spack_data_home` and not `spack_xdg_data_home`

So I think this comment should be closed